### DISCUSSION TITLE
Is their a way to sum / calculate a value for a chart without making a View?

### DISCUSSION THREAD

**Discussion by KernOil** [Budibase/budibase#2986] [Help]
Hi, I have been trying and failing to use the bar graph, to sum up, how many pages a drafter has been assigned.
I am building a dynamic dashboard for the drafters that will display their data based on their username.

I have a number field that the number of pages is noted in.  for the cards, I was able to use filters on a data provider to get the correct dynamic values. however, for the graphs, it looks like I am only able to sum a value over many rows in a view in the data table. This will not let the value change based on a date filter. Does anyone have any thoughts on a workaround?

**Comment by mjashanks** [comment]
Could you provide some more detail?...

1. What are the columns in your table?
2. What do you want to display on your X and Y axis of the chart?
3. Does the bar chart display data specific to the current user ?
4. What do you want your date filters to do? I.e. which columns should they filter on?

**Comment by KernOil** [comment]
Ok, the two columns I would like to use are the "Assigned to", and "number of drawings".  
the first is a name column the second is a number column. 
I would like to have the total number of drawings summed over a specific name and a specific date.

on my X-axis I would like the "assigned to" name on the Y-axis I would like the summed total of all of the pages assigned to that user over a given time.   

Yes, my chart displays data differently depending on what user is logged in. 

I would like the date to be used so the user can filter and see workload in different date frame's

![image](https://user-images.githubusercontent.com/80058238/137161736-6e054a93-59fc-4ae8-9337-89957035f4fd.png)
 This is how I would like it to look. I can achieve this with a view in the data table however date filtering is not available. 

![image](https://user-images.githubusercontent.com/80058238/137161854-3ef2a5d1-dd33-4e77-a97c-8dcbb63611aa.png)
this is what I get when using "define filters" for the data. date filtering works but as you can see each bar is a different request. I would like all of the bars summed into one.

**Reply by mjashanks** [reply]
Thanks for this - had a discussion about this. Views are quite simplistic at the moment and are due an overhaul.

It's just not possible to do what you're asking at the moment, but It's a really clear and common use case.

I've written up some notes here, please feel free to contribute your thoughts: https://github.com/Budibase/budibase/discussions/2999

**Reply by mjashanks** [reply]
PS - i know you said "Without using a view" - but I think Views should be the way to do this, even though we cannot.

**Reply by KernOil** [reply]
Perfect. either way works for me. Thank you.
